nwbwidgets/timeseries.py

```diff
@@ -966,4 +942,485 @@ def update(
+            ]
+        )
+        self.children = [self.control, self.figure_widget]
+
+    def update_filter_menu(self, change):
+        selected_columns = self.select_filter_columns.value
+        if selected_columns != (None,):
+            selection_boxes = [
+                widgets.Dropdown(options=self.options_per_column[column], description=column)
+                for column in selected_columns
+            ]
+            self.filter_menu.children = tuple(selection_boxes)
+            self.filtering_active = True
+            self.filter_menu.layout.visibility = "visible"
+
+        else:
+            self.filtering_active = False
+            self.filter_menu.layout.visibility = "hidden"
+
+    def update_row_faceting(self, change):
+        non_selected_columns = list(set(self.columns_for_filtering).difference(self.select_filter_columns.value))
+
+        self.faceting_row_selection.options = list(non_selected_columns)
+        self.faceting_row_selection.value = None
+
+    def update_column_faceting(self, change):
+        non_selected_columns = list(set(self.columns_for_filtering).difference(self.select_filter_columns.value))
+        self.faceting_column_selection.options = list(non_selected_columns)
+        self.faceting_column_selection.value = None
+
+    def query_expresion(self, children):
+        if isinstance(children.value, str):
+            return f" {children.description} == '{children.value}' "
+        else:
+            return f" {children.description} == {children.value} "
+
+    def update_plot_widget(self, button_instance):
+        # Load and prepare data
+
+        if self.filtering_active:
+            query_string = "and".join([self.query_expresion(children) for children in self.filter_menu.children])
+            df_query = self.trials_table.to_dataframe().query(query_string)
+        else:
+            df_query = self.trials_table.to_dataframe()
+
+        self.trialized_data_df = trialize_time_series(
+            time_series=self.time_series,
+            trials_table=df_query,
+            start_time_shift=self.start_time_shift_widget.value,
+            duration=(self.end_time_widget.value - self.start_time_shift_widget.value),
+            data_column=self.data_column_selection.value,
+            alignment_column=self.column_to_align_to_widget.value,
+        )
+
+        # Generate the plot
+        facet_col = self.faceting_column_selection.value
+        facet_row = self.faceting_row_selection.value
+        figure = build_faceting_figure(df=self.trialized_data_df, facet_col=facet_col, facet_row=facet_row)
+
+        # Update the widget
+        with self.figure_widget.batch_update():
+            self.figure_widget.update(layout_annotations=None)
+            self.figure_widget.update(figure.to_dict(), overwrite=True)
+            self.figure_widget.update_xaxes(matches=matches)
+
+
+def route_trialized_time_series(time_series: TimeSeries, neurodata_vis_spec=None, **kwargs):
```

Comment:
There is no routing going on here..

Reply:
It would also be good if you subclassed `TrializedTimeSeries` to have proper labels for specific modalities

Reply:
I was thinking that would be the role of the routing function. Initializing the visualization -for lack of a better term- to match the neurodata type that is specified for it in the visualization spec dictionary.

Reply:
This is done in the last commit and in the routing function.